model/pdata/common.go

```diff
@@ -379,7 +379,8 @@ func (a AttributeValue) AsString() string {
 		return strconv.FormatBool(a.BoolVal())
 
 	case AttributeValueTypeDouble:
-		return strconv.FormatFloat(a.DoubleVal(), 'f', -1, 64)
+		jsonStr, _ := json.Marshal(a.DoubleVal())
+		return string(jsonStr)
```

Comment:
Why? Should we use what json.Marshal uses instead? The reason is I believe json.Marshal is extremely slow because of reflection etc.

Reply:
We can do that.  It would involve copying https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/encoding/json/encode.go;l=585.

Reply:
Done. Benchmark improvement:

With `json.Marshal`:

`BenchmarkAttributeValueFloat_AsString-16    	 3270664	       364.8 ns/op`

With the new implementation:

`BenchmarkAttributeValueFloat_AsString-16    	 6703143	       174.0 ns/op`

I'll let you decide if the change is worth it :).

Reply:
As most likely a future maintainer of the codebase what would be your preference 😁?

Reply:
I think its probably worth it.  We risk drifting from the upstream implementation, but it hasn't changed in years, so we are probably safe.  The benchmark differences are fairly large.